### PR TITLE
Use `config.get` to access all `config` properties

### PR DESCRIPTION
**Pull request by GHaberis**
## What does this change?

Updates all points where we directly access `config` properties to use our `config.get` function. EG. `config.page.assetsPath` becomes `config.get('page.assetsPath')`.

Updates all related `.spec.js` files to use `config.set` rather than directly setting the properties on the `config`.

I've made this change because it'll help us track what config properties are being accessed by the commercial script running on `dotcom-rendering`.

## Screenshots

N/A

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/master/docs/03-dev-howtos/15-updating-test-database.md)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/dfp/dfp-env.js [hunk 1]
```diff
@@ -53,6 +53,6 @@ export const dfpEnv: DfpEnv = {
     shouldLazyLoad() {
         // We do not want lazy loading on pageskins because it messes up the roadblock
         // Also, if the special dll parameter is passed with a value of 1, we don't lazy load
-        return !!config.get('page.hasPageSkin') && getUrlVars().dll !== '1';
+        return !config.get('page.hasPageSkin') && getUrlVars().dll !== '1';
```
Comment (katebee) [inline static/src/javascripts/projects/commercial/modules/dfp/dfp-env.js]: We have lost a `!` here... Not sure if this is important / intentional?

File: static/src/javascripts/projects/commercial/modules/third-party-tags/simple-reach.spec.js [hunk 2]
```diff
@@ -3,6 +3,12 @@ import { simpleReach } from './simple-reach';
 
 const { shouldRun, url } = simpleReach;
 
+/**
+ * we have to mock config like this because
+ * loading simple-reach has side affects
+ * that are dependent on config.
+ * */
+
```
Comment (katebee) [inline static/src/javascripts/projects/commercial/modules/third-party-tags/simple-reach.spec.js]: 👏 👏 Thanks for documenting!

File: static/src/javascripts/projects/common/modules/onward/breaking-news.spec.js [hunk 3]
```diff
@@ -17,13 +17,29 @@ jest.mock('lib/storage', () => ({
...
-        pageId: '12345',
-        edition: 'UK',
-        section: 'football',
-    },
-}));
+/**
+ * we have to mock config like this because
+ * loading simple-reach has side affects
+ * that are dependent on config.
+ * */
```
Comment (katebee) [inline static/src/javascripts/projects/common/modules/onward/breaking-news.spec.js]: Comment may want updating; this is the breaking-news Spec 😉

Comment (GHaberis) [reply static/src/javascripts/projects/common/modules/onward/breaking-news.spec.js]: hah, yep that was a copy and paste mistake, good spot

**Comment by SiAdcock** [comment]
Great stuff @GHaberis! Are we close to updating [our ESLint config](https://github.com/guardian/frontend/blob/master/static/src/javascripts/.eslintrc.js#L82) to `error` instead of `warn` on direct access of `window.guardian.config`?

**Comment by GHaberis** [comment]
> Great stuff @GHaberis! Are we close to updating [our ESLint config](https://github.com/guardian/frontend/blob/master/static/src/javascripts/.eslintrc.js#L82) to `error` instead of `warn` on direct access of `window.guardian.config`?

OK I've had a go at doing this and it introduced a whole set of interesting challenges about the "errors" the rule picks up as not all should be errors, it also flagged up a whole load of new errors I'd not previously picked up in this PR. I intent to raise a follow up PR where I cleanup the remaining places we directly access `config` as this PR is big enough as it is.

**Review by SiAdcock — APPROVED** [review]
👍 Let's `config.get(🚢)`

**Pull request merged**